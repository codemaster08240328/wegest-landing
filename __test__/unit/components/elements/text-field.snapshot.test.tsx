import React from 'react'
import renderer from 'react-test-renderer'
import TextField from '@/components/elements/text-field'

describe('<TextField> snapshot', () => {
  it('renders correctly when there are no props', () => {
    expect(renderer.create(<TextField />).toJSON()).toMatchSnapshot()
  })

  it('renders correctly when there is icon', () => {
    expect(
      renderer.create(<TextField icon='email' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders correctly when there are icon and icolor', () => {
    expect(
      renderer.create(<TextField icon='email' icolor='primary' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders correctly when there is label', () => {
    expect(
      renderer.create(<TextField label='Email' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders correctly when there is className', () => {
    expect(
      renderer.create(<TextField className='class-1' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders correctly when there is value', () => {
    expect(
      renderer.create(<TextField value='Hello' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders correctly when there is helper text', () => {
    expect(
      renderer.create(<TextField helperText='Some important text' />).toJSON(),
    ).toMatchSnapshot()
  })
})
