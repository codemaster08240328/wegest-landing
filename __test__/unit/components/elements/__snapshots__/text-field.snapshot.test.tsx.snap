// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField> snapshot renders correctly when there are icon and icolor 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="class"
>
  <span
    aria-hidden={true}
    className="material-icons MuiIcon-root makeStyles-icon-2 MuiIcon-colorPrimary"
    data-testid="icon"
  >
    email
  </span>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-input-3"
    data-testid="text"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField> snapshot renders correctly when there are no props 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="class"
>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-input-3"
    data-testid="text"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField> snapshot renders correctly when there is className 1`] = `
<div
  className="makeStyles-root-1 class-1"
  data-testid="class"
>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-input-3"
    data-testid="text"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField> snapshot renders correctly when there is helper text 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="class"
>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-input-3"
    data-testid="text"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
    <p
      className="MuiFormHelperText-root"
    >
      Some important text
    </p>
  </div>
</div>
`;

exports[`<TextField> snapshot renders correctly when there is icon 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="class"
>
  <span
    aria-hidden={true}
    className="material-icons MuiIcon-root makeStyles-icon-2 MuiIcon-colorPrimary"
    data-testid="icon"
  >
    email
  </span>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-input-3"
    data-testid="text"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField> snapshot renders correctly when there is label 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="class"
>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-input-3"
    data-testid="text"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      Email
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField> snapshot renders correctly when there is value 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="class"
>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-input-3"
    data-testid="text"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="Hello"
      />
    </div>
  </div>
</div>
`;
